src/main/java/core/basesyntax/service/reader/ReadFileFruitImpl.java

```
@@ -0,0 +1,23 @@
+package core.basesyntax.service.reader;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.LinkedList;
+import java.util.List;
+
+public class ReadFileFruitImpl implements ReadFileFruit {
+    @Override
+    public List<String> readAll(String filePath) {
+        List<String> list = new LinkedList<>();
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(filePath))) {
+            String line;
+            while ((line = bufferedReader.readLine()) != null) {
+                list.add(line);
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);

```

The catch block rethrows the exception as `new RuntimeException(e)` which lacks an informative message. This violates the checklist: "Don't ignore exceptions in try-catch blocks\nDon't leave `e.printStackTrace()` or `System.out.println(e.getMessage())` in the catch blocks.\nLet's rethrow a RuntimeException with an **informative** message and exception object." Please rethrow with a clear message that includes the `filePath` and wrap the original exception, e.g.: `throw new RuntimeException("Can't read file by path: " + filePath, e);`. See catch block at lines 18–19.